### PR TITLE
[PLAT-767] implement storage and extrinsics

### PR DESCRIPTION
**Pull request by cowboy-bebug**
## Description

Re: [PLAT-767](https://ajunanetwork.atlassian.net/browse/PLAT-767)

## Type of changes

- [ ] `build`: Changes that affect the build system or external dependencies (eg, Cargo, Docker)
- [ ] `ci`: Changes to CI configuration
- [ ] `docs`: Changes to documentation only
- [ ] `feat`: Changes to add a new feature
- [ ] `fix`: Changes to fix a bug
- [ ] `refactor`: Changes that do not alter functionality
- [ ] `style`: Changes to format the code
- [ ] `test`: Changes to add missing tests or correct existing tests

## Checklist

- [ ] Tests for the changes have been added
- [ ] Necessary documentation is added (if appropriate)
- [ ] Formatted with `cargo fmt --all`
- [ ] Linted with `cargo clippy --all-features --all-targets`
- [ ] Tested with `cargo test --workspace --all-features --all-targets`

[PLAT-767]: https://ajunanetwork.atlassian.net/browse/PLAT-767?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by darkfriend77 — CHANGES_REQUESTED** [review]
frozen state and prepare_avatar seem to be redundant?
avatars can still be changed when in prepare_avatar storage?
proper cleanup when nft is created is missing?
prepare_fee properly taken?

File: pallets/ajuna-awesome-avatars/src/lib.rs
[the remark this answers is not in the document]
Comment (cowboy-bebug) [reply pallets/ajuna-awesome-avatars/src/lib.rs:1]: 👍 addressed in https://github.com/ajuna-network/Ajuna/pull/250

File: pallets/ajuna-awesome-avatars/src/lib.rs [hunk 1]
```diff
@@ -769,6 +804,9 @@ pub mod pallet {
 			ensure!(Self::global_configs().nft_transfer.open, Error::<T>::NftTransferClosed);
 			Self::ensure_unlocked(&avatar_id)?;
 
+			ensure!(Preparation::<T>::contains_key(avatar_id), Error::<T>::NotPrepared);
+			ensure!(FrozenAvatars::<T>::contains_key(avatar_id), Error::<T>::NotFrozen);
+
 			Self::do_transfer_avatar(&player, &Self::technical_account_id(), &avatar_id)?;
 			let collection_id = Self::collection_id().ok_or(Error::<T>::CollectionIdNotSet)?;
 			T::NftHandler::store_as_nft(player, collection_id, avatar_id, avatar)?;
```
[the remark this answers is not in the document]
Comment (cowboy-bebug) [reply pallets/ajuna-awesome-avatars/src/lib.rs:812]: 👍 addressed in https://github.com/ajuna-network/Ajuna/pull/250

File: pallets/ajuna-awesome-avatars/src/tests.rs
[the remark this answers is not in the document]
Comment (cowboy-bebug) [reply pallets/ajuna-awesome-avatars/src/tests.rs:1]: 👍 addressed in https://github.com/ajuna-network/Ajuna/pull/250

[the remark this answers is not in the document]
Comment (cowboy-bebug) [reply pallets/ajuna-awesome-avatars/src/tests.rs:1]: https://github.com/ajuna-network/Ajuna/pull/250/files

File: pallets/ajuna-awesome-avatars/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cowboy-bebug) [reply pallets/ajuna-awesome-avatars/src/lib.rs:809]: Created a follow up task [PLAT-781](https://ajunanetwork.atlassian.net/browse/PLAT-781)

**Review by darkfriend77 — DISMISSED** [review]
nice!

**Pull request merged**